devicetypes/smartthings/zwave-rgbw-controller.src/zwave-rgbw-controller.groovy

```diff
@@ -0,0 +1,241 @@
+			}
+			tileAttribute ("device.color", key: "COLOR_CONTROL") {
+				attributeState "color", action:"setColor"
+			}
+		}
+  }
+  controlTile("whiteSliderControl", "device.whiteLevel", "slider", width: 2, height: 2, inactiveLabel: false) {
+    state "whiteLevel", action:"setWhiteLevel", label:'White Level'
+ 	 }
+	standardTile("reset", "device.reset", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+		state "default", label:"Reset Color", action:"reset", icon:"st.lights.philips.hue-single"
+  	}
+	standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+		state "default", label:"Refresh", action:"refresh.refresh", icon:"st.secondary.refresh"
+	}
+
+	main(["switch"])
+	details(["switch", "levelSliderControl", "rgbSelector", "whiteSliderControl", "reset", "refresh"])
+}
+
+def updated() {
+     // Device-Watch simply pings if no device events received for 122 min (checkInterval)
+     log.debug("Adding checkInterval for health checks")
+     sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+     response(refresh())
+}
+
+def installed() {
+     // Device-Watch simply pings if no device events received for 122 min (checkInterval)
+     log.debug("Adding checkInterval for health checks")
+     sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def parse(description) {
+	def result = null
+	if (description.startsWith("Err 106")) {
+		state.sec = 0
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description, [0x20: 1, 0x26: 3, 0x70: 1, 0x33:3])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+			log.debug("'$description' parsed to $result")
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	dimmerEvents(cmd)
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value, descriptionText: "$device.displayName was turned $value")]
+	if (cmd.value) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%")
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.hailv1.Hail cmd) {
+	response(command(zwave.switchMultilevelV1.switchMultilevelGet()))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x84: 1])
+	if (encapsulatedCommand) {
+		state.sec = 1
```

Comment:
you don't need this anymore, as secure/insecure is now indicated by an "s" in the zwaveinfo.zw object.

see:

`zwaveInfo.zw.endsWith("s")`

in a couple of our DTHs.

Reply:
So remove the state.sec and the entire result part, leaving just the "zwaveEvent(encapsulatedCommand)" from what I can tell looking at the lock?

Reply:
yep

Reply:
Ok.  How do you handle line 86 then which is assigning state.sec = 0?  Remote that since you'd never check it again?  I.e. remove 86 & 87 then change 88 to just a if statement?

Reply:
nm, since you never check state.sec against anything there isn't a real good reason for setting it to 0.  Removed